manifests/pod.yaml

```diff
@@ -1,34 +1,120 @@
 # This pod template is used in the Jenkins pipeline by the Kubernetes plugin
 
 apiVersion: v1
+kind: Template
+labels:
+  app: ${DISTRO}-coreos
+  template: ${DISTRO}-coreos-template
 metadata:
-  name: coreos-assembler
-kind: Pod
-spec:
-  containers:
-   - name: jnlp
-     image: jenkins-slave-base-centos7:latest
-     args: ['$(JENKINS_SECRET)', '$(JENKINS_NAME)']
-   - name: coreos-assembler
-     image: coreos-assembler:master
-     imagePullPolicy: Always
-     command: ['/usr/bin/dumb-init']
-     args: ['sleep', 'infinity']
-     volumeMounts:
-     - name: data
-       mountPath: /srv/
-     - name: duffy-key
-       mountPath: /var/run/secrets/kubernetes.io/duffy-key
-       readOnly: true
-     securityContext:
-       privileged: false
-  nodeSelector:
-    oci_kvm_hook: allowed
-  volumes:
-  - name: data
-    persistentVolumeClaim:
-      claimName: coreos-assembler-claim
-  - name: duffy-key
-    secret:
-      secretName: duffy.key
-      optional: true
+  annotations:
+    description: |-
+      Jenkins pipeline for CoreOS distributions.
+    iconClass: icon-jenkins
+    openshift.io/display-name: Fedora CoreOS Pipeline
+    openshift.io/documentation-url: https://github.com/coreos/fedora-coreos-pipeline
+    openshift.io/support-url: https://github.com/coreos/fedora-coreos-pipeline
+    openshift.io/provider-display-name: fedora CoreOS
+  name: coreos-certs
+parameters:
+  - description: Distribution name
+    name: DISTRO
+    value: fedora
+  - description: Name of the namespace to use
+    name: NAMESPACE
+    value: fedora-coreos
+  - description: PEM cert data to trust.
+    name: CERT_DATA
+  - description: minimum amount of memory
+    value: 512M
+    name: MIN_MEMORY
+  - description: maximum amount of memory
+    value: 1024M
+    name: MAX_MEMORY
+  - description: minimum amount of cpu
+    value: "1"
+    name: MIN_CPU
+  - description: maximum amount of cpu
+    value: "1"
+    name: MAX_CPU
+
+objects:
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: trust-ca-cert
+      namespace: ${NAMESPACE}
+    data:
+      cert: |
+        ${CERT_DATA}
```

Comment:
Assuming that we can't use an openshift template here, we should be able to move this ConfigMap definition to pipeline.yaml, right?

Reply:
I would agree with that.